httpcore/_sync/connection.py

```diff
@@ -99,7 +99,7 @@ def _connect(self, request: Request) -> NetworkStream:
         while True:
             try:
                 if self._uds is None:
-                    kwargs = {
+                    kwargs: Dict[str, Any] = {
```

Comment:
ah this is because kwargs is reassigned in the `else` clause and Trace takes a `dict` when it should take a `collections.abc.Mapping` - for variance support